java/query/query-core/src/main/java/sleeper/query/core/rowretrieval/LeafPartitionQueryExecutor.java

```diff
@@ -76,12 +76,13 @@ public CloseableIterator<Row> getRows(LeafPartitionQuery leafPartitionQuery) thr
         Schema tableSchema = tableProperties.getSchema();
         String compactionIteratorClassName = tableProperties.get(TableProperty.ITERATOR_CLASS_NAME);
         String compactionIteratorConfig = tableProperties.get(TableProperty.ITERATOR_CONFIG);
-        ConfigStringIterator compactionIterator;
-        ConfigStringIterator queryIterator;
+        String compactionFilters = tableProperties.get(TableProperty.FILTERS_CONFIG);
+        SortedRowIterator compactionIterator;
+        SortedRowIterator queryIterator;
 
         try {
-            compactionIterator = createIterator(tableSchema, objectFactory, compactionIteratorClassName, compactionIteratorConfig);
-            queryIterator = createIterator(tableSchema, objectFactory, leafPartitionQuery.getQueryTimeIteratorClassName(), leafPartitionQuery.getQueryTimeIteratorConfig());
+            compactionIterator = createIterator(tableSchema, objectFactory, compactionIteratorClassName, compactionIteratorConfig, compactionFilters);
+            queryIterator = createIterator(tableSchema, objectFactory, leafPartitionQuery.getQueryTimeIteratorClassName(), leafPartitionQuery.getQueryTimeIteratorConfig(), compactionFilters);
```

Comment:
This is applying the compaction time filters to both the compaction iterator and the query iterator. It's not necessary to apply the filter twice.